core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2337,10 +2344,8 @@ private[spark] class DAGScheduler(
         "checkpointing the RDD before repartition and try again."
     }
 
-    activeJobs.foreach(job => collectStagesToRollback(job.finalStage :: Nil))
-
     // The stages will be rolled back after checking
-    val rollingBackStages = HashSet[Stage](mapStage)
+    val rollingBackStages = HashSet[Stage]()
```

Comment:
The old `mapstage` is part of `stagesToRollback` as the `succeedingStages` in the `collectSucceedingStages` was including the `mapStage`.

Reply:
https://github.com/apache/spark/pull/50630/commits/dd141c585595b4e24d8b9ad43ef32df0c67d9110